intern/cycles/kernel/kernel_color.h

```diff
@@ -33,6 +33,645 @@ ccl_device float linear_rgb_to_gray(KernelGlobals *kg, float3 c)
   return dot(c, float4_to_float3(kernel_data.film.rgb_to_y));
 }
 
+ccl_device float float_lerp(float start, float end, float progress)
+{
+    return start + ((end - start) * progress);
+}
+
+ccl_device float3 wavelength_to_xyz(float wavelength)
+{
+    float4 wavelength_xyz_lookup[] = {
```

Comment:
I'm fully aware this is a terrible way to do this. Looking for guidance on the 'proper' way to use this sort of data.